core/node/libp2p/rcmgr_defaults.go

```diff
@@ -129,7 +129,9 @@ func createDefaultLimitConfig(cfg config.SwarmConfig) (limitConfig rcmgr.Concret
 		}
 
 		// Scale System.StreamsInbound as well, but use the existing ratio of StreamsInbound to ConnsInbound
-		partialLimits.System.StreamsInbound = rcmgr.LimitVal(maxInboundConns * int64(partialLimits.System.StreamsInbound) / int64(partialLimits.System.ConnsInbound))
+		if partialLimits.System.StreamsInbound != rcmgr.Unlimited {
```

Comment:
Would it make sense to change this check to `if partialLimits.System.StreamsInbound < 0` to protect against _all_ enums (including any future ones that are added)?

Reply:
that's the point, we don't ever want to include enums in this calculation right?

Reply:
I don't know, maybe there is a new enum in the future that will require some new logic.

Reply:
The same issue would occur in any language if you overload an int w/ multiple semantics depending on where a value lies in an interval.